tests/Dfc.CourseDirectory.Testing/TestData/TestData.CreateProvider.cs

```diff
@@ -22,8 +22,17 @@ public partial class TestData
             string courseDirectoryName = "",
             string alias = "",
             ProviderDisplayNameSource displayNameSource = default,
-            IEnumerable<CreateProviderContact> contacts = null)
+            IEnumerable<CreateProviderContact> contacts = null,
+            IReadOnlyCollection<Guid> tLevelDefinitionIds = null)
```

Comment:
As discussed previously, I'd rather not see entities bundled together like this on mocks - it adds complexity and causes confusion.

Reply:
The entire purpose of these methods is to make setting up test data as low ceremony as possible. It also gives us a place to maintain the T-Level providers have at least one T Level Definition assigned invariant .

Reply:
I do see where you're coming from but ultimately it creates a maintenance overhead and additional complexity IMO.